modules/apps/data-engine/data-engine-taglib/src/main/resources/META-INF/resources/data_layout_builder/new-js/components/rules/RuleItem.js

```diff
@@ -0,0 +1,213 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+import ClayButton from '@clayui/button';
+import ClayLabel from '@clayui/label';
+import ClayPanel from '@clayui/panel';
+import classNames from 'classnames';
+import {useForm, useFormState} from 'dynamic-data-mapping-form-renderer';
+import React from 'react';
+
+import DropDown from '../../../js/components/drop-down/DropDown.es';
+import {
+	forEachDataDefinitionField,
+	getDataDefinitionField,
+	getFieldLabel,
+	getOptionLabel,
+} from '../../../js/utils/dataDefinition.es';
+import {getLocalizedValue} from '../../../js/utils/lang.es';
+import {EVENT_TYPES} from '../../eventTypes';
+
+const ACTION_LABELS = {
+	autofill: Liferay.Language.get('autofill'),
+	calculate: Liferay.Language.get('calculate'),
+	enable: Liferay.Language.get('enable'),
+	require: Liferay.Language.get('require'),
+	show: Liferay.Language.get('show'),
+};
+
+const OPERATOR_LABELS = {
+	'belongs-to': Liferay.Language.get('belongs-to'),
+	contains: Liferay.Language.get('contains'),
+	'equals-to': Liferay.Language.get('equals-to'),
+	'is-empty': Liferay.Language.get('is-empty'),
+	'not-contains': Liferay.Language.get('not-contains'),
+	'not-equals-to': Liferay.Language.get('not-equals-to'),
+	'not-is-empty': Liferay.Language.get('not-is-empty'),
+};
+
+const Text = ({capitalize = false, children = '', lowercase = false}) => (
+	<span
+		className={classNames('pr-1', {
+			'text-capitalize': capitalize,
+			'text-lowercase': lowercase,
+		})}
+	>
+		{children}
+	</span>
+);
+
+export default function RuleItem({rule, toggleRulesEditorVisibility}) {
+	const {actions, conditions, logicalOperator, name: ruleName} = rule;
+	const dispatch = useForm();
+	const {dataDefinition} = useFormState({schema: ['dataDefinition']});
+	const name = getLocalizedValue(dataDefinition.defaultLanguageId, ruleName);
+
+	const dropDownActions = [
+		{
+			action: () => toggleRulesEditorVisibility(rule),
+			name: Liferay.Language.get('edit'),
+		},
+		{
+			action: () => {
+				const confirmed = confirm(
+					Liferay.Language.get('are-you-sure-you-want-to-delete-this')
+				);
+
+				if (confirmed) {
+					dispatch({
+						payload: rule.ruleEditedIndex,
+						type: EVENT_TYPES.RULE.DELETE,
+					});
```

Comment:
As you would have to remove `ruleEditedIndex` you can do this from `RulesSidebar` and add a callback which is done similarly in `RulesSidebar`.

Something like `onDelete`, having `on` as a callback prefix is much better than `toggle`, because this standard already exists in the native APIs (`onChange`, `onMouse`...), so it is easier to reading and interpreting that this is a callback and more consistent.

Reply:
That is true! I fixed it.